apps/mysql.go

```diff
@@ -15,13 +15,75 @@
 package apps
 
 import (
-	"fmt"
-	"strings"
+	"os"
 
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+
+	"fmt"
+	"strings"
 )
 
+type MetricsReceiverMySQL struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,hostname_port|file"`
+
+	Password  string `yaml:"password" validate:"omitempty"`
+	Username  string `yaml:"username" validate:"omitempty"`
+	Transport string `yaml:"transport" validate:"omitempty"`
```

Comment:
What are the valid values for `transport`? They should be validated (and documented in the docs).

MySQL usually calls this parameter [`protocol`](http://download.nust.na/pub6/mysql/doc/refman/5.4/en/connecting.html#option_general_protocol) and the valid values are `TCP`, `SOCKET`, `PIPE`, and `MEMORY`, but also client libraries usually infer the correct value so users don't normally need to specify it. (Can we do that?)

Reply:
Otel is pushing to try and use [`config.NetAddr`](https://pkg.go.dev/go.opentelemetry.io/collector/config/confignet) struct for connections, which uses Transport as the wording here. but I'll change it to protocol for the ops-agent level.

As for defaulting for the protocols and what those defaults should be, I've tested using `unix` and `tcp` for the values of transport since it's based off of [net.Dial](https://pkg.go.dev/net#Dial) for the [client](https://github.com/go-sql-driver/mysql) and those worked. Since the protocols don't seem to line up with the shell of MySql, perhaps the ones outlined by net.Dial are the ones we should be using.

Reply:
If the values are `config.NetAddr` transports, we can keep the field called `transport` - that way people won't expect to be able to use mysql protocol names. (Though if we can infer it from whether the address starts with `/` or not that would be even better. Not sure how we would support the Windows protocols but I'd guess the underlying library doesn't either.)

Reply:
ok then we should be good with the update to the validation I just pushed.

Reply:
Can we also add a `oneof` validation to https://github.com/GoogleCloudPlatform/ops-agent/blame/78981928d79a29f3f33da56b16fd577f8604d5cd/apps/mysql.go#L35? Similar to https://github.com/GoogleCloudPlatform/ops-agent/blob/bd6630930fa4a1d672e522963bc8d9badcb30304/confgenerator/logging_receivers.go#L118. Plus a related unit test for invalid `transport` value.

Reply:
Added.

Reply:
Thanks!